common/lib/plugins/failover/failover_plugin.ts

```diff
@@ -441,6 +433,17 @@ export class FailoverPlugin extends AbstractConnectionPlugin {
             throw new AwsWrapperError(Messages.get("Failover.unableToDetermineWriter"));
           }
 
+          const allowedHosts = this.pluginService.getHosts();
+          if (!allowedHosts.includes(writerHostInfo)) {
+            const failoverErrorMessage = Messages.get(
+              "Failover.newWriterNotAllowed",
+              writerHostInfo ? "<null>" : writerHostInfo.host,
+              logTopology(allowedHosts, "[Failover.newWriterNotAllowed]")
+            );
+            logger.error(failoverErrorMessage);
+            throw new FailoverFailedError(failoverErrorMessage);
+          }
+
           await this.pluginService.abortCurrentClient();
           await this.pluginService.setCurrentClient(result.client, writerHostInfo);
           logger.debug(Messages.get("Failover.establishedConnection", this.pluginService.getCurrentHostInfo()?.host ?? ""));
```

Comment:
Not relevant to the changes in this PR but the catch block doesn't check for `FailoverSuccessError`, so `this.failoverWriterFailedCounter.inc();` will be called even if failover has succeeded